ntbs-service/Services/AuthorizationService.cs

```diff
@@ -117,27 +137,8 @@ public async Task<IQueryable<Notification>> FilterNotificationsByUserAsync(Claim
             {
                 _filter = await GetUserPermissionsFilterAsync(user);
             }
-    
-            if (_filter.FilterByTBService)
-            {
-                notifications = notifications.Where(n => _filter.IncludedTBServiceCodes.Contains(n.Episode.TBServiceCode));
-            }
-            else if (_filter.FilterByPHEC)
-            {
-                // Having a method in LINQ clause breaks IQuaryable abstraction. We have to use inline expression over methods
-                notifications = notifications.Where(n => 
-                    (
-                        n.Episode.TBService != null && 
-                        _filter.IncludedPHECCodes.Contains(n.Episode.TBService.PHECCode)
-                    ) || (
-                        n.PatientDetails.PostcodeLookup != null && 
-                        n.PatientDetails.PostcodeLookup.LocalAuthority != null && 
-                        n.PatientDetails.PostcodeLookup.LocalAuthority.LocalAuthorityToPHEC != null && 
-                        _filter.IncludedPHECCodes.Contains(n.PatientDetails.PostcodeLookup.LocalAuthority.LocalAuthorityToPHEC.PHECCode)
-                    )
-                );
-            }
-            return notifications;
+
+            return notifications.Where(n => UserHasDirectRelationToNotification(n));

```

Comment:
As much as I love the re-use of the abstraction, there's a worry here that LINQ can't use this, which results in us fetching ALL the notifications into memory, and only filtering them there, as opposed to the current behaviour where this clause gets translated to SQL by LINQ, only fetching the notifications we want.
See comment here: https://github.com/PublicHealthEngland/ntbs_Beta/pull/334/files/b437dadceb0139635f0196110ee0c72fcf0479ae..77486bf2ef6e36f2494e42210ce5b42197387248#diff-cb6801aec27f9f0ec167f6c8142db9d0L127
As this is used on the homepage with multiple notifications, that performance gain is significant!

It might be worth double checking, but I don't think we can get away with the commonisation here - but it might be worth adding more explicit comments about it!